test/unittest.cpp

```diff
@@ -11,6 +12,10 @@ namespace duckdb {
 static bool test_force_storage = false;
 static bool test_force_reload = false;
 static bool test_memory_leaks = false;
+static bool summarize_failures = false;
+// this counter is for the order number of the failed test case in Failures Summary
+static size_t failures_summary_counter = 0;
```

Comment:
Same here - can we move all of this to a single class that holds all of the required state? The class can either be a singleton or just have static methods.